### ISSUE TITLE
[regression/7.0.81] ScrollView content with Buttons rendered offscreen do not respond on iOS

### ISSUE THREAD

**Issue by csa89** [dotnet/maui#14624]
This is related to issue #9236, the issue appear again on the latest MAUI release 7.0.81 and 7.0.200 on iOS only. Every button that doesn’t appear directly on the screen after render will never get clicked. It appear in the following 3 scenarios: button in a scrollview, bindable stacklayout and ListView. It is tested using iOS 13.* and 16.* using a simulator and real devices.

**Comment by NonameMissingNo** [comment]
Can confirm it happens on 7.0.81.

**Comment by Redth** [comment]
Added a repro here: https://github.com/Redth/MauiIssueReproBrowser/tree/main/MauiReproBrowser/Issues/14624

So far I'm unable to confirm the repro, the code is working ok for me.

**Comment by Redth** [comment]
Once thing I noticed in the original repro example in #9236 is that the event handler is an inline delegate.  I wonder if something is getting garbage collected here and so the click event never appears to fire.

In my repro I use a separate event method which may help.  Arguably it may be wise to keep a reference to all the buttons created around too, however I'd expect that will be done by the layout they are added to.

**Comment by csa89** [comment]
[MauiAppScrollIssue.zip](https://github.com/dotnet/maui/files/11321138/MauiAppScrollIssue.zip)
Attached is a reproducible scenario using a BindableLayout.  "await Task.Delay(3000)" or a real api call cause the issue.
Click on a Collapse button off-screen will not be clicked as per the below image:

![Screenshot_1682416218](https://user-images.githubusercontent.com/123084543/234241631-377d7433-7155-4e44-ad54-98a898bbf41c.png)

**Comment by mtopolewski** [comment]
Hi, I found a solution!

What worked for me was calling SizeToFit method on ScrollView's first subview (which is ScrollView's contentView as I understand):
```
using Microsoft.Maui.Handlers;
#if IOS
using UIKit;
#endif

namespace IosScrollviewMauiTestApp;

public class FixedScrollview : ScrollView
{
  public FixedScrollview()
  {
#if IOS
      ScrollViewHandler.Mapper.AppendToMapping(nameof(IScrollView.ContentSize), OnScrollViewContentSizePropertyChanged);
#endif
  }

#if IOS
  private void OnScrollViewContentSizePropertyChanged(IScrollViewHandler _, IScrollView __)
  {
      if (Handler?.PlatformView is not UIView platformUiView)
          return;
      
      if (platformUiView.Subviews.FirstOrDefault() is not { } contentView)
          return;
      
      contentView.SizeToFit();
  }
#endif
}
```
I added reproduction repo here: https://github.com/mtopolewski/IosScrollViewMauiTestApp along with above fix. To see fix in action just replace ScrollView with FixedScrollView and run it on iOS. I tested it on iPhone 14 Pro Max with iOS 16.4, hope it will also work for you

**Comment by cat0363** [comment]
The workaround suggested by @mtopolewski also worked for Issue #14795.
Thank you.

**Comment by hartez** [comment]
May be fixed by #16385

**Comment by BerndStoltefuss** [comment]
> Hi, I found a solution!
> 
> What worked for me was calling SizeToFit method on ScrollView's first subview (which is ScrollView's contentView as I understand):
> 
> ```
> using Microsoft.Maui.Handlers;
> #if IOS
> using UIKit;
> #endif
> 
> namespace IosScrollviewMauiTestApp;
> 
> public class FixedScrollview : ScrollView
> {
>   public FixedScrollview()
>   {
> #if IOS
>       ScrollViewHandler.Mapper.AppendToMapping(nameof(IScrollView.ContentSize), OnScrollViewContentSizePropertyChanged);
> #endif
>   }
> 
> #if IOS
>   private void OnScrollViewContentSizePropertyChanged(IScrollViewHandler _, IScrollView __)
>   {
>       if (Handler?.PlatformView is not UIView platformUiView)
>           return;
>       
>       if (platformUiView.Subviews.FirstOrDefault() is not { } contentView)
>           return;
>       
>       contentView.SizeToFit();
>   }
> #endif
> }
> ```
> 
> I added reproduction repo here: https://github.com/mtopolewski/IosScrollViewMauiTestApp along with above fix. To see the fix in action just replace ScrollView with FixedScrollView in MainPage.xaml and run it on iOS. I tested it on iPhone 14 Pro Max with iOS 16.4, hope it will also work for you.
> 
> Repro steps:
> 
> 1. Run the app on iOS: https://github.com/mtopolewski/IosScrollViewMauiTestApp
> 2. Toggle the switch to show the rest of the form
> 3. Try to click on one of the entries at the bottom of the form (ScrollView's content)
> 
> Expected bahaviour: Entries should be focusable so you can click on them and start typing. Actual behaviour: nothing happens, so you can't focus on these entries and start typing.
> 
> Workaround? Replace ScrollView with FixedScrollView in MainPage.xaml.

While this caused gui freezes for me in some of my pages, this modified approach works for me (even with expanding content). This should do until the .NET7 backport is released.
` public class FixedScrollview : ScrollView
    {

        public FixedScrollview()
        {
#if IOS
            this.PropertyChanged += FixedScrollview_PropertyChanged;
#endif
        }

        private void FixedScrollview_PropertyChanged(object sender, System.ComponentModel.PropertyChangedEventArgs e)
        {
            if ("Content".Equals(e.PropertyName) && this.Content != null)
            {
                this.Content.SizeChanged += Content_SizeChanged;
            }
        }

        private void Content_SizeChanged(object sender, EventArgs e)
        {
#if IOS
            if (this.Handler?.PlatformView is UIView platformUiView && platformUiView?.Subviews?.FirstOrDefault(p => p.Tag == 0x845fed) is { } contentView)
            {
                contentView?.SizeToFit();
            }
#endif
        }
`

**Comment by hartez** [comment]
> May be fixed by #16385

Confirmed.

**Comment by baaaaif** [comment]
hmm while the workaround from @mtopolewski worked in my case with 7.0.92,
now with 7.0.96 the workaround doesn't work anymore AND the scrollview resizing is broken :(

**Comment by BerndStoltefuss** [comment]
Works for me without workaround in 7.0.96.
Thanks

**Comment by danielkraut** [comment]
@BerndStoltefuss Your workaround works for more different pages than the previous one. But I have a question how did you come up with that `Tag == 0x845fed`? Can you link me some documentation about tag values and there usage? I am quite new to MAUI.

**Comment by BerndStoltefuss** [comment]
@danielkraut I looked at what the new code is doing to identify the correct subview/contentPanel (see https://github.com/dotnet/maui/blob/0f764ac98e1a0e0e31b797ec991f8bba896a7c9c/src/Core/src/Handlers/ScrollView/ScrollViewHandler.iOS.cs#L17). It's an internal tag value, so undocumented.

But as i said, for me 7.0.96 seems to work fine without the workaround.

**Comment by MichaelSL** [comment]
it doesn't work for me with NET8 rc2;
my button is in the scroll view and it's not clickable on iOS

**Comment by danielkraut** [comment]
@BerndStoltefuss I updated my personal work environment and you are right - the issue is fixed in 7.0.0.96. Updating our CI runners will be more difficult though.
